### PR TITLE
Add more docs to DragTraget.builder

### LINKED ISSUE #151448: DragTarget docs could be improved
I got confused reading DragTarget docs tonight.

So I've sent a PR to add docs for DragTarget.builder.
https://github.com/flutter/flutter/pull/151447

PRs require issues, so here is an issue.

### PR DESCRIPTION
**Pull request by eseidel**
I wrote these after having gone through the DragTarget docs
tonight and having been confused.

I also changed one reference to (deprecated) onWillAccept
to onWillAcceptWithDetails.

The other changes to remove ! are just from `dart fix` running
automatically in my editor.  It appears that the modern analyzer
is smart enough to not need the !.  I'm happy to revert those if
Flutter is intentionally not running `dart fix` on files.

Fixes https://github.com/flutter/flutter/issues/151448

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (aliasgar4558) [inline packages/flutter/lib/src/widgets/drag_target.dart:42]: Can we update the comment as below :

/// The `candidateData` argument contains the list of drag data that is hovering
/// over this [DragTarget] and that has passed [DragTarget.onWillAcceptWithDetails].
/// The `rejectedData` argument contains the list of drag data that is hovering over this 
/// [DragTarget] and that will not be accepted by the [DragTarget].

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (aliasgar4558) [inline packages/flutter/lib/src/widgets/drag_target.dart:640]: Can be updated to :

/// [onWillAccept] or [onWillAcceptWithDetails] is called when a draggable enters the target. 
/// If true, then the data will appear in `candidateData` else in `rejectedData`.

Comment (justinmc) [reply packages/flutter/lib/src/widgets/drag_target.dart:640]: I agree, this way is more clear that one of those methods should be null. Though I would add a comma after `candidateData`:

```dart
/// [onWillAccept] or [onWillAcceptWithDetails] is called when a draggable enters the target. 
/// If true, then the data will appear in `candidateData`, else in `rejectedData`.
```

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (justinmc) [inline packages/flutter/lib/src/widgets/drag_target.dart:644]: Duplicate "and" here.

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (justinmc) [inline packages/flutter/lib/src/widgets/drag_target.dart:647]: Super nit: "[List]"?

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (justinmc) [inline packages/flutter/lib/src/widgets/drag_target.dart:651]: To [avoid saying "you"](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#use-the-passive-voice-recommend-do-not-require-never-say-things-are-simple):

"If unexpected `null` values appear in `candidateData` or `rejectData`,..."

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (dkwingsmt) [inline packages/flutter/lib/src/widgets/drag_target.dart:638]:
```suggestion
  /// The [onWillAccept] or [onWillAcceptWithDetails] is called when a draggable
```

Per https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#use-correct-grammar

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (justinmc) [reply packages/flutter/lib/src/widgets/drag_target.dart:42]: Did you just reformat it or am I missing what the change is?

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (dkwingsmt) [inline packages/flutter/lib/src/widgets/drag_target.dart:640]:
```suggestion
  /// `candidateData`, if both return false, the data will appear in
```

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (justinmc) [inline packages/flutter/lib/src/widgets/drag_target.dart:785]: Maybe your IDE is smarter than the analyzer, it seems like it needs the `!` to pass.

**Comment by dkwingsmt** [comment]
Omitting the `!` seems to have caused analyze errors. I guess it can't be removed after all.

  error • The argument type 'Offset?' can't be assigned to the parameter type 'Offset'.  • packages/flutter/lib/src/widgets/drag_target.dart:749:123 • argument_type_not_assignable
  error • The argument type 'Offset?' can't be assigned to the parameter type 'Offset'.  • packages/flutter/lib/src/widgets/drag_target.dart:785:94 • argument_type_not_assignable
  error • The argument type 'Offset?' can't be assigned to the parameter type 'Offset'.  • packages/flutter/lib/src/widgets/drag_target.dart:793:79 • argument_type_not_assignable
  error • The argument type 'Offset?' can't be assigned to the parameter type 'Offset'.  • packages/flutter/lib/src/widgets/drag_target.dart:963:48 • argument_type_not_assignable

**Comment by dkwingsmt** [comment]
test-exempt: documentation change with no semantic changes

**Review by justinmc — COMMENTED** [review]
Thanks for the PR Eric! Usually when I see your name on GitHub it's something that's like 6 years old so I had to double check :)

LGTM if you just get the analyzer passing, everything else is nits.

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (eseidel) [reply packages/flutter/lib/src/widgets/drag_target.dart:42]: I changed one mention of (deprecated) DragTarget.onWillAccept to DragTarget.onWillAcceptWithDetails and re-wrapped.

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (eseidel) [reply packages/flutter/lib/src/widgets/drag_target.dart:638]: Does it need to be "The xxx method" then?  I'm not sure how "The" helps, but I also don't have any strong opinions happy to defer to those more active in this section of the code!

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (eseidel) [inline packages/flutter/lib/src/widgets/drag_target.dart:793]:
```suggestion
    widget.onMove?.call(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!));
```

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (eseidel) [inline packages/flutter/lib/src/widgets/drag_target.dart:785]:
```suggestion
      widget.onAcceptWithDetails?.call(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!));
```

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (eseidel) [inline packages/flutter/lib/src/widgets/drag_target.dart:963]:
```suggestion
    onDragEnd?.call(velocity ?? Velocity.zero, _lastOffset!, wasAccepted);
```

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (eseidel) [inline packages/flutter/lib/src/widgets/drag_target.dart:749]:
```suggestion
                                    widget.onWillAcceptWithDetails!(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!)));
```

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (eseidel) [inline packages/flutter/lib/src/widgets/drag_target.dart:748]:
```suggestion
                                    widget.onWillAcceptWithDetails!(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!)));
```

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (eseidel) [inline packages/flutter/lib/src/widgets/drag_target.dart:784]:
```suggestion
      widget.onAcceptWithDetails?.call(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!));
```

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (eseidel) [inline packages/flutter/lib/src/widgets/drag_target.dart:792]:
```suggestion
    widget.onMove?.call(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!));
```

File: packages/flutter/lib/src/widgets/drag_target.dart
Comment (eseidel) [inline packages/flutter/lib/src/widgets/drag_target.dart:962]:
```suggestion
    onDragEnd?.call(velocity ?? Velocity.zero, _lastOffset!, wasAccepted);
```

**Comment by eseidel** [comment]
Fixes applied, PTAL.

**Review by justinmc — APPROVED** [review]
LGTM :+1:

**Pull request merged**
